cookbook/security/entity_provider.rst

```diff
@@ -5,69 +5,49 @@
+you how to load your users from the database via a Doctrine entity.
 
 Introduction
 ------------
 
-This article focuses on how to authenticate users against a database table
-managed by a Doctrine entity class. The content of this cookbook entry is split
-in three parts. The first part is about designing a Doctrine ``User`` entity
-class and making it usable in the security layer of Symfony. The second part
-describes how to easily authenticate a user with the Doctrine
-:class:`Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider` object
-bundled with the framework and some configuration.
-Finally, the tutorial will demonstrate how to create a custom
-:class:`Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider` object to
-retrieve users from a database with custom conditions.
+Before you start, you should check out the `FOSUserBundle`_. This external
+bundle allows you to load users from the database (like you'll learn here)
+*and* gives you built-in routes and controllers for common things like login,
+registration and forgot password. Whether you use that bundle or go through
+the steps here is up to you.
 
-.. sidebar:: Code along with the Example
+Loading users via a Doctrine entity has 2 basic steps:
 
-    If you want to follow along with the example in this chapter, create
-    an AcmeUserBundle via:
+1. :ref:`Create your User entity <security-crete-user-entity>`
+2. :ref:`Configure security.yml to load from your entity <security-config-entity-provider>`
 
-    .. code-block:: bash
+Afterwards, you can learn more about :ref:`forbidding inactive users <security-advanced-user-interface>`,
+:ref:`using a custom query <authenticating-someone-with-a-custom-entity-provider>`
+and :ref:`user serialization to the session <cookbook-security-serialize-equatable>`
 
-        $ php app/console generate:bundle --namespace=Acme/UserBundle
+.. _security-crete-user-entity:
+.. _the-data-model:
 
-The Data Model
---------------
+1) Create your User Entity
+--------------------------
 
-For the purpose of this cookbook, the AcmeUserBundle bundle contains a ``User``
-entity class with the following fields: ``id``, ``username``, ``password``,
-``email`` and ``isActive``. The ``isActive`` field tells whether or not the
-user account is active.
-
-To make it shorter, the getter and setter methods for each have been removed to
-focus on the most important methods that come from the
-:class:`Symfony\\Component\\Security\\Core\\User\\UserInterface`.
-
-.. tip::
-
-    You can :ref:`generate the missing getter and setters <book-doctrine-generating-getters-and-setters>`
-    by running:
-
-    .. code-block:: bash
-
-        $ php app/console doctrine:generate:entities Acme/UserBundle/Entity/User
+For this entry, suppose that you already have a ``User`` entity inside an
+``AppBundle`` with the following fields: ``id``, ``username``, ``password``,
+``email`` and ``isActive``:
```

Comment:
what about replacing this long section with `php app/console doctrine:generate:entity --entity="AppBundle:User" --fields="..."` ?